corehq/motech/utils.py

```diff
@@ -144,8 +148,11 @@ def reencrypt_ecb_to_cbc_mode(encrypted_text, existing_prefix=None):
         ciphertext = encrypted_text[len(existing_prefix):]
     else:
         ciphertext = encrypted_text
-
-    new_ciphertext = b64_aes_cbc_encrypt(b64_aes_decrypt(ciphertext))
+    try:
+        plaintext = b64_aes_decrypt(ciphertext)
+    except UnicodeDecodeError:
+        raise AesEcbDecryptionError("Failed to decrypt the AES-ECB-encrypted text.")
```

Comment:
Just curious, in what case we might fail to decrypt it?

Reply:
I'm not sure why. On staging (and only staging, not prod) `ConnectionSettings` contains encrypted passwords that after being decrypted, raises a `UnicodeDecodeError` when attempting to decode it. But since it's already unusable (so probably hasn't been used), I'm raising this error so I can set the new value to `''` during the migration.

Reply:
Did you print those encrypted passwords, are they start with the ECB prefix? I'm worried something goes wrong when we encrypt the ConnectionSettings pwd.

Reply:
the existing passwords that were failing do start with the `aes` prefix.

> I'm worried something goes wrong when we encrypt the ConnectionSettings pwd.

I'm not following - what's the concern you have?

Reply:
Reread your previous reply again, I think what you said make sense to me! Thank you.